### PR TITLE
Fixes #1090

### LINKED ISSUE #1090: Auto-publish Ruby/Yard docs
We need a process by which every release publishes the Yard docs or links them from the README.

### PR DESCRIPTION
**Pull request by ramkumar-kr**
- Add link for documentation published by rubygems in README
- Remove rake task to generate YARD docs

### CODE REVIEW

**Comment by dblock** [comment]
Needs changelog please, also remove yard from Gemfile (and elsewhere?)

**Comment by ramkumar-kr** [comment]
Done

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #### Features
 
 * [#1686](https://github.com/ruby-grape/grape/pull/1686): Avoid coercion of a value if it is valid - [@timothysu](https://github.com/timothysu).
+* [#1688](https://github.com/ruby-grape/grape/pull/1688): Fixes #1090 - [@ramkumar-kr](https://github.com/ramkumar-kr).
```
Comment (dblock) [inline CHANGELOG.md]: Lets be explicit here, make it "Removed Yard docs".

Comment (ramkumar-kr) [reply CHANGELOG.md]: done

**Comment by dblock** [comment]
Thanks

**Pull request merged**